### PR TITLE
authentication added

### PR DESCRIPTION
**Pull request by margarita-ignatieva**

### CODE REVIEW

**Review by IrynaDim — COMMENTED** [review]
Maybe we need to set role for new user during registration in register method (AuthenticationImpl class).

File: src/main/java/com/cinema/shop/dao/impl/RoleDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl implements RoleDao {
+    private static final Logger log = Logger.getLogger(OrderDaoImpl.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/shop/dao/impl/RoleDaoImpl.java]: OrderDaoImpl ?

File: src/main/java/com/cinema/shop/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> userOptional = userService.findByEmail(email);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/shop/security/CustomUserDetailsService.java]: let's use orElseThrow here instead, that will make code shorter

File: src/main/java/com/cinema/shop/dao/impl/RoleDaoImpl.java (hunk 1 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/dao/impl/RoleDaoImpl.java]: nope, haven't changed

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**